docs/basics/101-114-filesystem.rst

```diff
@@ -0,0 +1,127 @@
+operations, such as copying, moving, or renaming a files or
+directories.
+
+If I renamed on of those books, would the symlink that points
+to the file content still be correct? What happens if I'd copy
+an annexed file?
+If I moved the whole ``books/`` directory? What if I moved
+all of ``DataLad-101`` into a different place on my computer?
+What if renamed the whole superdataset?
+And how do I remove a file, or directory, or subdataset?
+
+There is no better way of learning than doing. Here, in the
+safe space of the ``DataLad-101`` course, you can try out all
+of the things you would be unsure about or reluctant to try
+on the dataset that contains your own, valuable data.
+
+What happens if I rename an annexed file?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's try it. In Unix, renaming a file is exactly the same as
+moving a file.
+
+.. runrecord:: _examples/DL-101-114-101
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   cd books/
+   mv TLCL.pdf The_Linux_Command_Line.pdf
+   ls -lah
+
+Try to open the renamed file, e.g. with
+``evince The_Linux_Command_Line.pdf``.
+This works!
+
+But let's see what changed in the dataset with this operation:
+
+.. runrecord:: _examples/DL-101-114-102
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ datalad status
+
+We can see that the old file is marked as ``deleted``, and
+simultaneously, an ``untracked`` file appears: the renamed
+PDF.
+
+In theory, we would need to do a ``datalad save`` at this point
+to commit the "deletion" and the "new file". This however, is
+suboptimal, and Git has built-in commands that provide a better
+solution.
+
+Lets revert the renaming of the the files:
+
+.. runrecord:: _examples/DL-101-114-103
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ git reset --hard master       # this restores TLCL.pdf
+   $ rm The_Linux_Command_Line.pdf       # this removes the untracked copy
+
+A more convenient way to rename files is the ``git mv`` command:
+
+.. runrecord:: _examples/DL-101-114-104
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ git mv TLCL.pdf The_Linux_Command_Line.pdf
+
+.. runrecord:: _examples/DL-101-114-105
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ datalad status
+
+We can see that the old file is still seen as "deleted", but the "new",
+renamed file is "added". A ``git status`` displays the change
+in the dataset a bit more accurate:
+
+.. runrecord:: _examples/DL-101-114-106
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ git status
+
+.. todo::
+
+   can I mitigate a ``git commit`` here? I believe the best ``datalad save``
+   command to save the renaming (without the rest of the dataset state) is
+   ``datalad save -m "renamed TLCL to verbose title" The_Linux_Command_Line.pdf TLCL.pdf``,
+   but this still requires listing all paths, and tab completion does not seem to work for
+   the "deleted" files...
```

Comment:
No, after a `git mv`, a `git commit` is the most elegant way. However, there is no need for `git mv`, as Git itself doesn't really record the "move". Instead, it tends to infer things post-hoc from the recorded commits. Watch this:

```
$ datalad create demo
$ cd demo
$ echo 123 > one
$ datalad save
# plain move, not Git
$ mv one two
$ datalad save
$ git log --patch HEAD~1..HEAD
commit 4c39756a94928a529d9000b573a10ddf75991c01 (HEAD -> master)
Author: Michael Hanke <michael.hanke@gmail.com>
Date:   Tue Jul 16 13:01:01 2019 +0200

    [DATALAD] Recorded changes

diff --git a/one b/two
similarity index 100%
rename from one
rename to two
```

Reply:
It is a bit trickier to `mv` a file into a subdirectory:

```
$ mkdir subdir
$ mv two subdir/two
```

The symlink will be broken. In any case, a `datalad save` also rectifies that.

In contrast to the above, Git will not detect the rename. Why? Because the content that is tracked in Git is the actual symlink, and due to the change in location, the nature of the symlink has to change. Hence, what looks like a move on the filesystem is actually a move plus a content change for Git.